torchrl/envs/transforms/transforms.py

```diff
@@ -4696,15 +4696,18 @@ def _reset(
         spec shape is assumed to match the tensordict's.
 
         """
-        shape = (
-            ()
-            if (not self.parent or self.parent.batch_locked)
-            else tensordict.batch_size
-        )
         _reset = _get_reset(self.reset_key, tensordict)
         if _reset.any():
             for key, spec in self.primers.items(True, True):
+                if spec.shape[: len(tensordict.batch_size)] != tensordict.batch_size:
```

Comment:
I found it necessary to add that. So primer can be defined with independence of the batch size of the env.